app/models/assignment.rb

```diff
@@ -20,23 +20,9 @@ class Assignment < Assessment
   # Add assignment_properties to default scope because we almost always want to load an assignment with its properties
   default_scope { includes(:assignment_properties) }
 
-  has_many :rubric_criteria,
+  has_many :criteria,
            -> { order(:position) },
-           class_name: 'RubricCriterion',
-           dependent: :destroy,
-           inverse_of: :assignment,
-           foreign_key: :assessment_id
-
-  has_many :flexible_criteria,
-           -> { order(:position) },
-           class_name: 'FlexibleCriterion',
-           dependent: :destroy,
-           inverse_of: :assignment,
-           foreign_key: :assessment_id
-
-  has_many :checkbox_criteria,
-           -> { order(:position) },
-           class_name: 'CheckboxCriterion',
+           class_name: 'Criterion',
```

Comment:
Hmm I don't think this line should be necessary any more, actually. Can you try removing it?